src/storage/src/hummock/local_version.rs

```diff
@@ -203,10 +203,14 @@ impl LocalVersion {
     pub fn seal_epoch(&mut self, epoch: HummockEpoch, is_checkpoint: bool) {
         // TODO: remove it when non-checkpoint barrier is enabled
         assert!(is_checkpoint, "current seal_epoch must be a checkpoint");
-        self.sealed_epoch = self.sealed_epoch.max(epoch);
-        if self.advance_max_sync_epoch(epoch).is_none() {
-            tracing::trace!("trivial advance max sync epoch: {}", epoch);
-        }
+        assert!(
+            epoch > self.sealed_epoch,
+            "sealed epoch not advance. new epoch: {}, current {}",
+            epoch,
+            self.sealed_epoch
+        );
+        self.sealed_epoch = epoch;
+        self.advance_max_sync_epoch(epoch)
```

Comment:
After non-checkpoint barrier is introduced, this will become:
```
if is_checkpoint {
  self.advance_max_sync_epoch(epoch)
}
```
right?

Reply:
Correct.